doc/sphinx/how-to/how-to-enable-graphics-core22-on-a-device.md

```diff
@@ -0,0 +1,632 @@
+VK_LAYER_PATH=${VK_LAYER_PATH:+$VK_LAYER_PATH:}${SELF}/share/vulkan/implicit_layer.d/:${SELF}/share/vulkan/explicit_layer.d/
+XDG_DATA_DIRS=${XDG_DATA_DIRS:+$XDG_DATA_DIRS:}${SELF}/share
+
+
+# These are in the default LD_LIBRARY_PATH, but in case the snap dropped it inadvertently
+if [ -d "/var/lib/snapd/lib/gl" ] && [[ ! ${LD_LIBRARY_PATH} =~ (^|:)/var/lib/snapd/lib/gl(:|$) ]]; then
+  LD_LIBRARY_PATH=${LD_LIBRARY_PATH}:/var/lib/snapd/lib/gl
+fi
+
+if [ -d "/var/lib/snapd/lib/glvnd/egl_vendor.d" ]; then
+  # This needs to be prepended, as glvnd goes depth-first on these
+  __EGL_VENDOR_LIBRARY_DIRS=/var/lib/snapd/lib/glvnd/egl_vendor.d:${__EGL_VENDOR_LIBRARY_DIRS}
+fi
+
+if [ -d "/var/lib/snapd/lib/vulkan/icd.d" ]; then
+  XDG_DATA_DIRS=${XDG_DATA_DIRS}:/var/lib/snapd/lib
+fi
+
+if [ -d "/var/lib/snapd/lib/gl/vdpau" ]; then
+  LD_LIBRARY_PATH=${LD_LIBRARY_PATH}:/var/lib/snapd/lib/gl/vdpau
+fi
+
+export LD_LIBRARY_PATH
+# Mesa-specific environment variable pointing to directory containing $VENDOR_dri.so files
+export LIBGL_DRIVERS_PATH
+export LIBVA_DRIVERS_PATH
+# glvnd-specific environment variable pointing to directory containing ICD descriptor files
+export __EGL_VENDOR_LIBRARY_DIRS
+# EGL external platform interface specific environment
+export __EGL_EXTERNAL_PLATFORM_CONFIG_DIRS
+export VK_LAYER_PATH
+export XDG_DATA_DIRS
+
+exec "$@"
+```
+
+This script has exported the proper paths via environment variables so that
+any snap that connects to our provider snap can properly resolve these
+libraries.
+
+### i386 Support
+If you also need to provide your snap for `i386`, you will need to provide
+a few extra things in the `graphics-core22-provider-wrapper`.
+An example from `mesa-core22` adds these lines:
+
+```sh
+# /bin/graphics-core22-provider-wrapper
+
+...
+if [ "$SNAP_ARCH" == "amd64" ]; then
+  GCONV_PATH=${GCONV_PATH:+$GCONV_PATH:}${SELF}/lib/i386-linux-gnu/gconv
+fi
+
+if [ -d "/var/lib/snapd/lib/gl32" ] && [[ ! ${LD_LIBRARY_PATH} =~ (^|:)/var/lib/snapd/lib/gl32(:|$) ]]; then
+  LD_LIBRARY_PATH=${LD_LIBRARY_PATH}:/var/lib/snapd/lib/gl32
+fi
+
+if [ -d "/var/lib/snapd/lib/gl32/vdpau" ]; then
+  LD_LIBRARY_PATH=${LD_LIBRARY_PATH}:/var/lib/snapd/lib/gl32/vdpau
+fi
+
+...
+
+[ -z ${GCONV_PATH+x} ] || export GCONV_PATH
+```
+
+The `snapcraft.yaml` file will also need to be updated to point at the 32 bit paths.
+You may find an example of using `i386` in 
+[mesa-core22](https://github.com/canonical/mesa-core22/blob/9d64e64fb06f7052f9e6f8a8899cac763f5fad7e/snap/snapcraft.yaml#L176).
+
+
+### How to supply the vendor libraries
+The following example supplies drivers from `libmali`. Using a parameter for
+`make`, the libraries will be installed to `$SNAPCRAFT_PART_INSTALL/usr/lib`.
+These libraries do not need to be in a particular path.
+
+
+```yaml
+parts:
+  ...
+  vendor-drivers:
+  plugin: make
+  source: libmali
+  make-parameters:
+    - LIBDIR=$SNAPCRAFT_PART_INSTALL/usr/lib
```

Comment:
```suggestion
    - LIBDIR=$SNAPCRAFT_PART_INSTALL/usr/lib/$CRAFT_ARCH_TRIPLET_BUILD_FOR
```

This will put it into `/usr/lib/arm64-linux-gnu`, which is where the wrapper script we demonstrate will look for it.